test/langtools/tools/javac/reflect/TestIRFromAnnotation.java

```diff
@@ -55,12 +55,20 @@
 import javax.lang.model.element.*;
 import javax.tools.JavaCompiler;
 import javax.tools.JavaFileManager;
-import javax.tools.JavaFileObject;
 import javax.tools.SimpleJavaFileObject;
 import javax.tools.ToolProvider;
 
 public class TestIRFromAnnotation {
 
+    static final Set<String> EXCLUDED_TEST = Set.of(
```

Comment:
I've added a list of tests to skip. First, trying to scan a testng test with this annotation processor doesn't work because certain testng dependencies cannot be resolved (but this is just noise in the test output).
More importantly, doing annotation processing on a method that contains local classes yields inconsistent local class names. Note that a local class name depend on _all_ the local classes contained within that compilation unit (e.g. if there's 3 methods, each declaring a class `Foo`, you get `Outer$1Foo`, `Outer$2Foo` and `Outer$3Foo`.
But the annotation processing code doesn't know that: whenever we retrieve the model for a method being compiled, we type-check the method on the fly, which means we generate "transient" inner class names, which ignore any other method. I think this is an intrinsic limitation on how annotation processors do things (in fact it would be much cleaner if we could extract the code model from a method being compiler _after_ type-checking, e.g. using a compiler plugin rather than an annotation processor. That's how null checker frameworks prefer to do things).

This will need to be addressed separately.